tests/core/pyspec/eth2spec/test/helpers/fork_transition.py

```diff
@@ -0,0 +1,142 @@
+    state_transition_and_sign_block,
+    transition_to,
+)
+from eth2spec.test.helpers.block import (
+    build_empty_block_for_next_slot,
+    build_empty_block,
+    sign_block,
+)
+
+
+def _state_transition_and_sign_block_at_slot(spec,
+                                             state,
+                                             operation_dict=None):
+    """
+    Cribbed from ``transition_unsigned_block`` helper
+    where the early parts of the state transition have already
+    been applied to ``state``.
+
+    Used to produce a block during an irregular state transition.
+    """
+    block = build_empty_block(spec, state)
+    # we can't just pass `body` and assign it because randao_reveal and eth1_data was set in `build_empty_block`
+    # thus use dict to pass operations.
+    if operation_dict is not None:
+        for key, value in operation_dict.items():
+            setattr(block.body, key, value)
+
+    assert state.latest_block_header.slot < block.slot
+    assert state.slot == block.slot
+    spec.process_block(state, block)
+    block.state_root = state.hash_tree_root()
+    return sign_block(spec, state, block)
+
+
+def _all_blocks(_):
+    return True
+
+
+def skip_slots(*slots):
+    """
+    Skip making a block if its slot is
+    passed as an argument to this filter
+    """
+    def f(state_at_prior_slot):
+        return state_at_prior_slot.slot + 1 not in slots
+    return f
+
+
+def no_blocks(_):
+    return False
+
+
+def only_at(slot):
+    """
+    Only produce a block if its slot is ``slot``.
+    """
+    def f(state_at_prior_slot):
+        return state_at_prior_slot.slot + 1 == slot
+    return f
+
+
+def state_transition_across_slots(spec, state, to_slot, block_filter=_all_blocks):
+    assert state.slot < to_slot
+    while state.slot < to_slot:
+        should_make_block = block_filter(state)
+        if should_make_block:
+            block = build_empty_block_for_next_slot(spec, state)
+            signed_block = state_transition_and_sign_block(spec, state, block)
+            yield signed_block
+        else:
+            next_slot(spec, state)
+
+
+def state_transition_across_slots_with_ignoring_proposers(spec, state, to_slot, ignoring_proposers):
+    """
+    The slashed validators can't be proposers. Here we ignore the given `ignoring_proposers`
+    and ensure that the result state was computed with a block with slot >= to_slot.
+    """
+    assert state.slot < to_slot
+
+    found_valid = False
+    while state.slot < to_slot or not found_valid:
+        future_state = state.copy()
+        next_slot(spec, future_state)
+        proposer_index = spec.get_beacon_proposer_index(future_state)
+        if proposer_index not in ignoring_proposers:
+            block = build_empty_block_for_next_slot(spec, state)
+            signed_block = state_transition_and_sign_block(spec, state, block)
+            yield signed_block
+            if state.slot >= to_slot:
+                found_valid = True
```

Comment:
seems like you can drop the variable `found_valid` if you just `break` once the condition is met

```suggestion
                break
```

Reply:
I want in the last, to have one block of `slot >= to_slot`.

If I remove `or not found_valid` from `while state.slot < to_slot or not found_valid`, the while loop might return without the last block.